README.rst

```diff
@@ -793,6 +793,31 @@ When you install this package, a couple command line tools are included and inst
 - `HouseCanary Analytics API Export <housecanary/hc_api_export>`_
 - `HouseCanary API Excel Concat <housecanary/hc_api_excel_concat>`_
 
+Running Tests
+---------------------------
+
+To run the unit test suite:
+
+::
+
+    python setup.py nosetests --with-coverage --cover-package=housecanary
+
+During unit tests, all API requests are mocked.
+You can run the unit tests with real API requests by doing the following:
+
+
+- Update `URL_PREFIX` in `constants.py` to point to a test or dev environment
+- Obtain an account for that environment that has permissions to all API components
+- Obtain an API Key and Secret and put them in `HC_API_KEY` and `HC_API_SECRET` environment variables
+- Run the test suite as: 
+
+::
+
+    HC_API_CALLS=true python setup.py nosetests
```

Comment:
Ah thanks. Didn't have that diff loaded.

You'll need to [write and register](http://nose.readthedocs.io/en/latest/plugins/writing.html) a custom nose plugin. These articles also made it more clear: https://community.activestate.com/impatient-developers-guide-writing-python-nose-plugins http://mahmoudimus.com/blog/2011/02/developing-a-nose-test-plugin-to-time-python-tests/

So you could name it 'live-api-calls' and the command would be '--with-live-api-calls'. Off hand, I don't see a super clean way to get that config all the way to your `mock_requests` function. Maybe still set the environment variable.

I also found out that when running the tests, it will fail if the api key environment variables aren't set. Line 41 in api_client.py. You could set some dummy environment variable in the nose plugin to take care of that, too.